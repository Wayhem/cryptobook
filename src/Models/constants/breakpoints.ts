export enum Breakpoints {
  tablet = 1024,
  mobile = 768,
}
