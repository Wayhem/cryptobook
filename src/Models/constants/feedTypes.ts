enum FeedTypes {
  Book_snapshot = 'book_ui_1_snapshot',
  standard = 'book_ui_1',
}

export default FeedTypes
