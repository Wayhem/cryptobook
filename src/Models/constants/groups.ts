enum XBTUSD_GROUPS_ENUM {
  small = 0.5,
  medium = 1,
  large = 2.5,
}

const XBTUSD_GROUPS = [XBTUSD_GROUPS_ENUM.small, XBTUSD_GROUPS_ENUM.medium, XBTUSD_GROUPS_ENUM.large]

export { XBTUSD_GROUPS, XBTUSD_GROUPS_ENUM }
