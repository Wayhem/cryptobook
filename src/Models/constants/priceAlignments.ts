enum Alignments {
  left = 'left',
  right = 'right',
}

export default Alignments
