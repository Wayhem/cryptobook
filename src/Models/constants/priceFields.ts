const fields = ['price', 'size', 'total']

export default fields
