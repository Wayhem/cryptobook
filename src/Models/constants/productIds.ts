enum ProductIds {
  PI_XBTUSD = 'PI_XBTUSD',
  PI_ETHUSD = 'PI_ETHUSD',
}

export default ProductIds
