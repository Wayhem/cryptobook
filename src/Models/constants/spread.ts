const spread = 12

export default spread
