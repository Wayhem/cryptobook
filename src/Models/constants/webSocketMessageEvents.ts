enum WSMessageEvents {
  subscribe = 'subscribe',
  unsubscribe = 'unsubscribe',
}

export default WSMessageEvents
