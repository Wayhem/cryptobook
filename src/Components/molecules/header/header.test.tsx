import Header from 'Components/molecules/header'
import { XBTUSD_GROUPS_ENUM } from 'Models/constants/groups'
import ProductIds from 'Models/constants/productIds'
import { render } from 'Utils/test-utils'

describe('Header page component', () => {
  test('Header renders and matches snapshot', () => {
    const defaultProps = {
      title: 'Hello',
      group: XBTUSD_GROUPS_ENUM.small,
      setGroup: jest.fn(),
      productId: ProductIds.PI_XBTUSD,
    }

    const { container } = render(<Header {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })
})
