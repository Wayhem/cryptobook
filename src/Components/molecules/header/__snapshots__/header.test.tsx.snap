// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header page component Header renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bdfBQB dycULH"
  >
    <div
      class="sc-gsTEea dRwGRi"
    >
      Hello
    </div>
    <div>
      <label
        class="sc-hKgJUU TebGZ"
      >
        group:
      </label>
      <select
        class="sc-dlfnuX gNzwCX"
      >
        <option
          value="0.5"
        >
          0.5
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2.5"
        >
          2.5
        </option>
      </select>
    </div>
  </div>
</div>
`;
